containers/src/Kokkos_ScatterView.hpp

```diff
@@ -774,6 +774,11 @@ class ScatterView<DataType, Layout, DeviceType, Op, ScatterNonDuplicated,
   ScatterView(View<RT, RP...> const& original_view)
       : internal_view(original_view) {}
 
+  template <typename RT, typename... P, typename... RP>
+  ScatterView(execution_space const& /* exec_space */,
+              View<RT, RP...> const& original_view)
+      : internal_view(original_view) {}
```

Comment:
```C++
static_assert(is_execution_space<execution_space>::value, "");

Reply:
We're getting `execution_space` from the class's own typedef, that extracts it from `DeviceType` on line 757. If it's not valid, it would seem that the relevant assertions should be at the level of the class, not any one constructor.

Reply:
Oh you are right.  Never mind.